src/pybind/mgr/prometheus/module.py

```diff
@@ -1104,6 +1124,33 @@ def add_fixed_name_metrics(self):
 
         self.metrics.update(new_metrics)
 
+    @profile_method(True)
```

Comment:
why profile this very method?

Reply:
Sake of completeness, I guess. I don't have a strong opinion on this one.To be precise, also the `collect` method doesn't completely fit into the requirement of profiling metric gathering methods, but in case there's a significant difference between the time it took to run `collect` and the sum of all other methods, I'd be able to deduce where the problem is (something in the `collect` method would have then been called which isn't profiled separately).

Reply:
Forget about what I said. It's because the @profile_method also logs the time it took to run a method. And `get_collect_time_metrics` is deliberately excluded from the evaluation inside the `get_collect_time_metrics` method.

Reply:
`collect()` calls, for instance `get_health()`. which in turn calls into `MgrModule.get()`. it might takes relatively long time to populate the data into a python object and then serialize it into a JSON string. 

but `get_collect_time_metrics()` just check the `_execution_duration ` property attached methods, and then set its value to `self.metrics`. I don't think it'd take a long time as long as the number of methods is bounded.

Reply:
>  And `get_collect_time_metrics` is deliberately excluded from the evaluation inside the `get_collect_time_metrics` method.

I don't think this addresses my question. It explains how you addresses the introduced problem which could have been avoided if we just remove `@profile_method(True)` here.

Reply:
It was to be on the safe side, really. It's true, there's no reason for it that it should take a significant amount of time, though, as it's part of `collect` I wanted the time it takes to run it to be logged (which is what `@profile_method(True)`) also does, but then excluded it from being evaluated as it wouldn't fit into the scheme of the resulting metrics. However, I have no strong opinion for keeping it around. As you already stated, it does not contain code that could potentially take longer than expected. I'm gonna remove it.